packages/table/src/headers/editableName.tsx

```diff
@@ -23,37 +23,55 @@ export interface IEditableNameProps extends IIntentProps, IProps {
      * important to listen to if you are doing anything with `onChange` events,
      * since you'll likely want to revert whatever changes you made.
      */
-    onCancel?: (value: string) => void;
+    onCancel?: (value: string, columnIndex?: number) => void;
 
     /**
      * A listener that is triggered as soon as the editable name is modified.
      * This can be due, for example, to keyboard input or the clipboard.
      */
-    onChange?: (value: string) => void;
+    onChange?: (value: string, columnIndex?: number) => void;
 
     /**
      * A listener that is triggered once the editing is confirmed. This is
      * usually due to the `return` (or `enter`) key press.
      */
-    onConfirm?: (value: string) => void;
+    onConfirm?: (value: string, columnIndex?: number) => void;
 }
 
 @PureRender
 export class EditableName extends React.Component<IEditableNameProps, {}> {
     public render() {
-        const { className, intent, name, onCancel, onChange, onConfirm } = this.props;
+        const { className, intent, name } = this.props;
         return (
             <EditableText
                 className={classNames(className, Classes.TABLE_EDITABLE_NAME)}
                 defaultValue={name}
                 intent={intent}
                 minWidth={null}
-                onCancel={onCancel}
-                onChange={onChange}
-                onConfirm={onConfirm}
+                onCancel={this.handleCancel}
+                onChange={this.handleChange}
+                onConfirm={this.handleConfirm}
                 placeholder=""
                 selectAllOnFocus={true}
             />
         );
     }
+
+    private handleCancel = (value: string) => {
+        this.invokeCallback(this.props.onCancel, value);
+    };
+
+    private handleChange = (value: string) => {
+        this.invokeCallback(this.props.onChange, value);
+    };
+
+    private handleConfirm = (value: string) => {
+        this.invokeCallback(this.props.onConfirm, value);
+    };
+
+    private invokeCallback(callback: (value: string, rowIndex?: number, columnIndex?: number) => void, value: string) {
```

Comment:
These callback params look inconsistent with the ones in the props interface above.

In the props interface: `value, columnIndex`
Here: `value, rowIndex, columnIndex`